src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -411,6 +411,120 @@ nlohmann::json SysInfo::getNetworks() const
     return networks;
 }
 
+std::string getProcessName(const std::string filePath)
+{
+    // Get stat file content.
+    std::string processInfo { UNKNOWN_VALUE };
+    const std::string statContent {Utils::getFileContent(filePath)};
+
+    const auto openParenthesisPos {statContent.find("(")};
+    const auto closeParenthesisPos {statContent.find(")")};
+
+    if (openParenthesisPos != std::string::npos && closeParenthesisPos != std::string::npos)
+    {
+        processInfo = statContent.substr(openParenthesisPos + 1, closeParenthesisPos - openParenthesisPos - 1);
+    }
+
+    return processInfo;
+}
+
+int64_t findInode(const std::string filePath)
+{
+    const size_t MAX_LENGTH {256};
+    char buffer[MAX_LENGTH];
+    const auto ret {readlink(filePath.c_str(), buffer, MAX_LENGTH)};
+    int64_t inode {-1};
+
+    // ret format is "socket:[<num>]".
+    if (-1 != ret)
+    {
+        const auto openBracketPos {std::string(buffer).find("[")};
+        const auto closeBracketPos {std::string(buffer).find("]")};
+        const auto match {std::string(buffer).substr(openBracketPos + 1, closeBracketPos - openBracketPos - 1)};
+        int64_t matchN;
+
+        try
+        {
+            matchN = std::stoi(match);
+            inode = matchN;
+        }
+        catch (...)
+        {}
+    }
+
+    return inode;
+}
+
+ProcessInfo portProcessInfo(const std::string& procPath, const std::vector<int64_t>& inodes)
+{
+    ProcessInfo ret;
+
+    if (Utils::existsDir(procPath))
+    {
+        std::vector<std::string> procFiles = Utils::enumerateDir(procPath);
+
+        // Iterate proc directory.
+        for (const auto& procFile : procFiles)
+        {
+            // Only directories that represent a PID are inspected.
+            const std::string procFilePath {procPath + "/" + procFile};
+
+            if (Utils::isNumber(procFile) && Utils::existsDir(procFilePath))
+            {
+                std::vector<std::string> pidFiles = Utils::enumerateDir(procFilePath);
+
+                // Iterate PID directory.
+                for (const auto& pidFile : pidFiles)
```

Comment:
What is the use of iterating the whole "/proc/{proc_number}" directory if we are only interested in the "fd" subdirectory?
Maybe we can ask if the directory "/proc/{proc_number}/fd" exists and then work with it.

Reply:
Touche!